### DISCUSSION TITLE
Update the macOS app icon?

### DISCUSSION THREAD

**Discussion by jackhinkle** [utmapp/UTM#3212] [Ideas]
Hey.

I put macOS icons together as a hobby of sorts. It’s not exactly as easy as making a single raster image and scaling it down to each required size. At this point, after several months, I made a template that I am 95% certain can produce a direct match to official production icons. Size, scaling, shadows, everything.

I decided to make an icon for UTM. I had a few issues with the existing icon:
- The shadows on the UTM logo array/waffle thing are very harsh.
- The smaller sizes are not correctly optimized.

Here’s what I came up with for the general design of the icon:
![Icon](https://user-images.githubusercontent.com/58039705/139069564-c6101798-9f2e-4b68-9109-190cfe56ffc7.png)
[Backup Imgur link for the design](https://i.imgur.com/64sUoLe.png)
The shadows are definitely not as harsh, but they’re still present… somewhat.

Apple’s official [Human Interface Guidelines for app icons](https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/app-icon/) states:
’As icon size decreases, fine details become muddy and hard to distinguish. At the smallest sizes, it’s important to remove unnecessary features and exaggerate primary features to help the content remain clear.’

The current icon doesn’t do this. So I made it do that. Here’s what I came up with:
![16x](https://user-images.githubusercontent.com/58039705/139069921-f37fc25b-c7fa-4c97-93a3-9d344d62426e.png)
[Backup Imgur link for the 16x icon](https://i.imgur.com/lzgj1dt.png)

![32x](https://user-images.githubusercontent.com/58039705/139069977-4a811795-9662-4a4b-ade2-38b006baada4.png)
Error: By 'Launchpad,' I mean Launchpad folder.
[Backup Imgur link for the 32x icon](https://i.imgur.com/nSI7sYA.png) 

I **had** to remove one of the rings for these smaller icons. There is no other way. Luckily, these are still undeniably UTM. I exported the icons at their native sizes for both my 2x and 1x display to make sure they look right and fit in. Again, these are using the official guidelines. Except for shadows. Apple is a liar and their Sketch template doesn’t have the correct shadows.

I’ve attached a zip file of the iconset so that the sizes can be previewed. The files in this are only for the purpose of viewing. Please do not use these in the project without further consent. I feel that this could be a good contribution to the project, although I’d like some feedback from some higher-ups in this repo first. If this follows through, I might decide to work on the iOS icons out of boredom next.

[iconset.zip](https://github.com/utmapp/UTM/files/7426277/iconset.zip)

**Comment by osy** [comment]
Thanks I like it. I’m not really a designer so I basically took the iOS icon and added shadows. I’m still not 100% sure what the difference should be between the iOS icon and the macos icon but it seems like apple goes for more shadows lol.

**Reply by kkebo** [reply]
There is a guideline.

https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/app-icon/

> **Use the drop shadow in the icon-design template.** The [template](https://developer.apple.com/design/resources/#macos-apps) includes the system-defined drop shadow that helps your app icon coordinate with other macOS 11 icons.

**Reply by jackhinkle** [reply]
@kkk669 
The provided templates do not have shadows that are accurate to official macOS icons.
![Artboard](https://user-images.githubusercontent.com/58039705/139099529-b6fdc93c-2cf7-4ab7-aee9-6f841529bd1e.png)
Here's an example. I took the Safari icon and content-awared it so the compass is gone. On the left is that official macOS icon, and on the right is one made from the official Sketch template. The shadows are different.

This has been an issue for me since the templates were originally released in 2020, which led to me spending months to just make my own template. The template that I made, which I used for this icon, was extensively referenced with official icons and their shadows. Refine, export, compare. Refine, export, compare. The template I made has accurate shadows within a single RGB unit.

**Reply by jackhinkle** [reply]
Alright. I might do a pull. I'm familiar with how Xcode handles icons, so it shouldn't be difficult. I actually did something similar [a while back](https://github.com/leits/MeetingBar/pull/155).

I've attached a zip that contains the source files, so if anybody in the future looks back at this and wants to tinker around with my template, go ahead.
[UTM.zip](https://github.com/utmapp/UTM/files/7427562/UTM.zip)

**Reply by osy** [reply]
Thanks

**Comment by jackhinkle** [comment]
Made a pull, finished.
